### PR TITLE
[6127] Insecure file upload turn on azure virus scanning

### PR DESCRIPTION
**Pull request by d-a-v-e**
### Context

Following on from [this ticket](https://trello.com/c/k6OnoVN1/6031-insecure-file-upload-medium) we should leverage azure’s defender for cloud to do virus scanning on uploads in our buckets.

### Changes proposed in this pull request

* adds MalwareScan service for uploads

### CODE REVIEW

File: app/jobs/malware_scanning_job.rb [hunk 1]
```diff
@@ -0,0 +1,10 @@
+# frozen_string_literal: true
+
+class MalwareScanningJob < ApplicationJob
+  def perform
+    Upload
+      .where(malware_scan_result: "pending")
+      .where("created_at < ?", 5.minutes.ago)
+      .find_each { |upload| MalwareScan.call(upload:) }
```
Comment (darokel) [inline app/jobs/malware_scanning_job.rb:8]: Could we fan these jobs out like we do in other areas?

File: app/services/malware_scan.rb [hunk 2]
```diff
@@ -0,0 +1,67 @@
+# frozen_string_literal: true
+
+class MalwareScan
```
Comment (darokel) [inline app/services/malware_scan.rb:3]: Should we put this in `app/lib` or namespace under `app/services/azure`?

File: db/migrate/20231002164646_add_malware_scan_result_to_uploads.rb [hunk 3]
```diff
@@ -0,0 +1,7 @@
+# frozen_string_literal: true
+
+class AddMalwareScanResultToUploads < ActiveRecord::Migration[7.0]
+  def change
+    add_column :uploads, :malware_scan_result, :string, default: "pending"
```
Comment (darokel) [inline db/migrate/20231002164646_add_malware_scan_result_to_uploads.rb:5]: all other enum fields appear to be setup as integer columns, should this be the same?

File: app/services/malware_scan.rb (hunk 2 above)
Comment (d-a-v-e) [reply app/services/malware_scan.rb:3]: I've moved it to `lib`

**Pull request merged**